tasks/check-data-keys/check-data-keys.yaml

```diff
@@ -67,4 +80,59 @@ spec:
             done
         done
 
+        exit $RC
+    - name: check-data-key-sources
+      image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+      script: |
+        #!/usr/bin/env sh
+        set -ex
+
+        DISALLOWED_KEYS_JSON='{
```

Comment:
So you'd like me to move this code to a step in `collect-data` instead? The whole point of having the data field be unstructured was so that the expected format isn't public, AIUI, so I don't think a config file in a yaml will be a good solution. I can also see value in allowing some fields to be defined in a few different places.. for example the releaseNotes.description could be in the Release or ReleasePlan. I see that as totally valid. 
Maintaining a list of allowed fields instead of a list of disallowed fields sounds more burdensome. I don't really see the advantage of it over defining the disallowed keys.

Reply:
I think this is something that should have been brought up in grooming the ticket

Reply:
> So you'd like me to move this code to a step in collect-data instead?

Yeah, I think that would make sense.

> The whole point of having the data field be unstructured was so that the expected format isn't public

Hmm, but it kind of is public here, no? It's just harder to read for everybody. And why shouldn't it be public?

> I can also see value in allowing some fields to be defined in a few different places..

What I'm suggesting would allow for that.

Right now I think it's not very easy for anybody to understand what the allowed/expected fields are without going through our code - like this check. Having a schema or some yaml config would make it much clearer I think.

Reply:
> I think this is something that should have been brought up in grooming the ticket

Good point. I don't recall grooming this, but I'm sure it happened :) I guess I didn't realize this then. But again, it's not something I feel very strong about.

Reply:
> > So you'd like me to move this code to a step in collect-data instead?
> 
> Yeah, I think that would make sense.
Okay, I can move it
> 
> > The whole point of having the data field be unstructured was so that the expected format isn't public
> 
> Hmm, but it kind of is public here, no? It's just harder to read for everybody. And why shouldn't it be public?
You'd have to ask Ralph or Brian. It was something about dirtying a CRD that non rh users will use with rh specific fields IIRC
> 
> > I can also see value in allowing some fields to be defined in a few different places..
> 
> What I'm suggesting would allow for that.
> 
> Right now I think it's not very easy for anybody to understand what the allowed/expected fields are without going through our code - like this check. Having a schema or some yaml config would make it much clearer I think.

I agree, but that is because our documentation sucks. I don't want any developer or user having to read tekton task bash code to figure out what fields go in CRs

Reply:
> Okay, I can move it

I'd like to hear what others think first.

> You'd have to ask Ralph or Brian. It was something about dirtying a CRD that non rh users will use with rh specific fields IIRC

I see. That makes sense. That makes sense. That's why I would put it outside of the CRD definition.

> I agree, but that is because our documentation sucks. I don't want any developer or user having to read tekton task bash code to figure out what fields go in CRs

Agreed.

Reply:
Adding @scoheb to the conversation too here about the validation. I think there is some talk of having a schema in some of the repos that hold the RPAs, but we still want validation in the pipelinerun itself, right?